be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -236,6 +692,26 @@ RowsetSharedPtr BetaRowsetWriter::build() {
             return nullptr;
         }
     }
+    Status status;
+    status = _wait_flying_segcompaction();
+    if (!status.ok()) {
+        LOG(WARNING) << "segcompaction failed when build new rowset, res=" << status;
+        return nullptr;
+    }
+    status = _segcompaction_ramaining_if_necessary();
+    if (!status.ok()) {
+        LOG(WARNING) << "segcompaction failed when build new rowset, res=" << status;
```

Comment:
We should not return error due to segcompaction failure?

Reply:
Theoretically, it should not. Non-deadly failures are handled as need-retry. Here is the the last-ditch for the remaining failures that is hard to handle by simply retrying. 

Eliminating them is hard but not impossible, as long as more effort had been made. I think we can polish them up in future patches if it causes any real problem.